File: webapi/Contracts/Endpoints/UserEndpoints.cs
﻿namespace webapi.Contracts.Endpoints
{
    public static class UserEndpoints
    {
        public const string BaseUrl = "api/User";

        public const string All = "All";

        public const string Usernames = "Usernames";

        public const string GetById = "{id:int}";

        public const string GetByUsername = "{username}";

        public const string Delete = "{id:int}";

        public const string Create = "";

        public const string Login = "Login";
    }
}

File: angularapp/src/app/shared/contracts/endpoints/UserEndpoints.ts
import { baseUrl } from '../../types';

const userBaseUrl = `${baseUrl}/user`;

export const UserEndpoints: Readonly<{
  all: string;
  usernames: string;
  getById: (id: number) => string;
  getByUsername: (username: string) => string;
  delete: (id: number) => string;
  create: string;
  login: string;
}> = {
  all: `${userBaseUrl}/all`,
  usernames: `${userBaseUrl}/usernames`,
  getById: (id: number) => `${userBaseUrl}/${id}`,
  getByUsername: (username: string) => `${userBaseUrl}/${username}`,
  delete: (id: number) => `${userBaseUrl}/${id}`,
  create: `${userBaseUrl}`,
  login: `${userBaseUrl}/login`,
};
